File: conversational-hub-flutter/lib/features/chat/domain/entities/message.dart
import 'package:equatable/equatable.dart';

enum MessageType { text, card, system }

class Message extends Equatable {
  final String id;
  final String channelId;
  final String senderId;
  final String content;
  final MessageType type;
  final Map<String, dynamic>? metadata;
  final DateTime timestamp;

  const Message({
    required this.id,
    required this.channelId,
    required this.senderId,
    required this.content,
    required this.type,
    this.metadata,
    required this.timestamp,
  });

  @override
  List<Object?> get props => [id, channelId, senderId, content, type, metadata, timestamp];
}

File: conversational-hub/src/main/java/com/allianceever/conversationalhub/domain/entities/Message.java
package com.allianceever.conversationalhub.domain.entities;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;
import java.util.Map;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Message {
    private String id;
    private String channelId;
    private String senderId;
    private String recipientId;
    private String content;
    private MessageType type;
    private Map<String, Object> metadata;
    private LocalDateTime timestamp;

    public enum MessageType {
        TEXT, CARD, SYSTEM
    }
}
